src/uu/du/src/du.rs

```diff
@@ -129,18 +130,39 @@ impl Stat {
             size: metadata.len(),
             blocks: size_on_disk / 1024 * 2,
             inode: file_info,
-            created: windows_time_to_unix_time(metadata.creation_time()),
+            created: windows_creation_time_to_unix_time(metadata.creation_time()),
             accessed: windows_time_to_unix_time(metadata.last_access_time()),
             modified: windows_time_to_unix_time(metadata.last_write_time()),
         })
     }
 }
 
 #[cfg(windows)]
-// https://doc.rust-lang.org/std/os/windows/fs/trait.MetadataExt.html#tymethod.creation_time
+// https://doc.rust-lang.org/std/os/windows/fs/trait.MetadataExt.html#tymethod.last_access_time
 // "The returned 64-bit value [...] which represents the number of 100-nanosecond intervals since January 1, 1601 (UTC)."
+// "If the underlying filesystem does not support last access time, the returned value is 0."
 fn windows_time_to_unix_time(win_time: u64) -> u64 {
-    win_time / 10_000_000 - 11_644_473_600
+    match (win_time / 10_000_000).checked_sub(11_644_473_600) {
+        None => 0,
+        Some(t) => t,
+    }
+}
+
+#[cfg(windows)]
+fn windows_creation_time_to_unix_time(win_time: u64) -> Option<u64> {
+    match (win_time / 10_000_000).checked_sub(11_644_473_600) {
+        None => None,
+        Some(t) => Some(t),
+    }
+}
+
+#[cfg(not(windows))]
+fn birth_u64(meta: &Metadata) -> Option<u64> {
+    meta.created()
+        .ok()
+        .and_then(|t| t.duration_since(UNIX_EPOCH).ok())
+        .map(|e| Some(e.as_secs() as u64))
+        .unwrap_or_else(|| None)
```

Comment:
Very minor suggestion: you could use `and_then` instead of `map` and remove `unwrap_or_else`. Also, when the closure of `unwrap_or_else` is just a literal, you might want to use `unwrap_or` instead.